pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -686,6 +689,9 @@ private BrokerResponseNative handleRequest(long requestId, String query,
     _queryLogger.log(
         new QueryLogger.QueryLogParams(requestId, query, requestContext, tableName, numUnavailableSegments, serverStats,
             brokerResponse, totalTimeMs, requesterIdentity));
+
+    Tracing.ThreadAccountantOps.clear();
```

Comment:
Should this be done in `finally` ?

Reply:
Yeah good catch, I have also seen a cleanup issue in test because of this.